bluemira/base/constants.py

```diff
@@ -42,68 +59,192 @@
 
 # Electron-volts to Joules
-EV_TO_J = 1.602176565e-19
+EV_TO_J = ureg.Quantity(1, ureg.eV).to("joule").magnitude
 
 # Joules to Electron-volts
-J_TO_EV = 1 / EV_TO_J
+J_TO_EV = ureg.Quantity(1, ureg.joule).to("eV").magnitude
 
 # Atomic mass units to kilograms
-AMU_TO_KG = 1.660539040e-27
+AMU_TO_KG = ureg.Quantity(1, ureg.amu).to("kg").magnitude
 
 # Years to seconds
-YR_TO_S = 60 * 60 * 24 * 365
+YR_TO_S = ureg.Quantity(1, ureg.year).to("s").magnitude
 
 # Seconds to years
-S_TO_YR = 1 / YR_TO_S
+S_TO_YR = ureg.Quantity(1, ureg.second).to("year").magnitude
+
+
+def to_celsius(kelvin: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    kelvin: Union[float, np.array, List[float]]
+        The temperature to convert [K]
+
+    Returns
+    -------
+    temp_in_celsius: Union[float, np.array]
+        The temperature [°C]
+    """
+    if np.any(np.less(kelvin, ABS_ZERO_K)):
+        raise ValueError("Negative temperature in K specified.")
+    return ureg.Quantity(kelvin, ureg.kelvin).to("celsius").magnitude
+
+
+def to_kelvin(celsius: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a temperature in Celsius to Kelvin.
+
+    Parameters
+    ----------
+    temp_in_celsius: Union[float, np.array, List[float]]
+        The temperature to convert [°C]
+
+    Returns
+    -------
+    temp_in_kelvin: Union[float, np.array]
+        The temperature [K]
+    """
+    if np.any(np.less(celsius, ABS_ZERO_C)):
+        raise ValueError("Negative temperature in K specified.")
+    return ureg.Quantity(celsius, ureg.celsius).to("kelvin").magnitude
+
+
+def kgm3_to_gcm3(density: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a density in kg/m3 to g/cm3
+
+    Parameters
+    ----------
+    density : Union[float, np.array, List[float]]
+        The density [kg/m3]
+
+    Returns
+    -------
+    density_gcm3 : Union[float, np.array]
+        The density [g/cm3]
+    """
+    return density / 1000.0
+
+
+def gcm3_to_kgm3(density: Union[float, np.array, List[float]]) -> Union[float, np.array]:
+    """
+    Convert a density in g/cm3 to kg/m3
+
+    Parameters
+    ----------
+    density : Union[float, np.array, List[float]]
+        The density [g/cm3]
+
+    Returns
+    -------
+    density_kgm3 : Union[float, np.array]
+        The density [kg/m3]
+    """
+    return density * 1000.0
+
+
+def pam3s_to_mols(flow_in_pam3_s):
+    """
+    Convert a flow in Pa.m^3/s to a flow in mols.
+
+    Parameters
+    ----------
+    flow_in_pam3_s: Union[float, np.array]
+        The flow in Pa.m^3/s to convert
+
+    Returns
+    -------
+    flow_in_mols: Union[float, np.array]
+        The flow in mol/s
+
+    Notes
+    -----
+    At 273.15 K for a diatomic gas
+    """
+    return flow_in_pam3_s / 2270
```

Comment:
I will try and find my workings for this gem... probably buried in a document somewhere, sorry!

Reply:
To be checked by Tiago

Reply:
@je-cook 
Hey James, here's the conversion you requested:
![Screenshot 2022-03-01 at 14 56 10](https://user-images.githubusercontent.com/40567154/156182209-c515fcc1-b93a-4271-8869-1004c31858b1.png)

Basically, [Pa.m3] can be converted to [mol] using a normalization at "normal temperature", which I had to check. But that's 0ºC and you already have that in the script (T_N = 273.15K). This makes the conversion factor to be:

8.3144598 * 273.15 ~ 2271.1

Reply:
Thanks very much!